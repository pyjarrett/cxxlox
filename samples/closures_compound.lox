fun foo() {
    var a = 10;
    var b = 20;

    fun bar() {
        fun qat() {
            print "qat";
            print a;
        }

        qat();
        print "bar";
        print a;
    }

    bar();
    print "foo";
    print a;
    print b;
}

foo();