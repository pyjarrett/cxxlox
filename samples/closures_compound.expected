qat
10
bar
10
foo
10
20
